login/service.go

```diff
@@ -316,59 +308,36 @@ func (keycloak *KeycloakOAuthProvider) CreateOrUpdateIdentityAndUser(ctx context
 		return &redirectTo, nil
 	}
 
-	// If the 'initlinking" param == true then initiate account linking if not already linked
-	linked, err := keycloak.checkAllFederatedIdentities(ctx, keycloakToken.AccessToken, brokerEndpoint)
-	if err != nil {
-		log.Error(ctx, map[string]interface{}{
-			"err": err,
-		}, "failed to check federated identities")
-		return nil, autherrors.NewInternalError(ctx, err)
-	}
-	log.Debug(ctx, map[string]interface{}{
-		"code":        code,
-		"referrerURL": referrerURL.String(),
-		"user_name":   identity.Username,
-		"linked":      linked,
-	}, "identities links checked")
+	redirectTo := referrerURL.String()
+	return &redirectTo, nil
+}
 
-	// Return linked=true param if account has been linked to all IdPs or linked=false if not.
-	if linked {
-		referrerStr := referrerURL.String() + "&linked=true"
-		redirectTo := referrerStr
-		log.Info(ctx, map[string]interface{}{
-			"referrerURL": referrerURL.String(),
-			"user_name":   identity.Username,
-			"linked":      linked,
-			"api_client":  apiClient,
-		}, "all good; redirecting back to referrer")
-		return &redirectTo, nil
+// AuthCodeURL is used in authorize action of /api/authorize to get authorization_code
+func (keycloak *KeycloakOAuthProvider) AuthCodeURL(ctx *app.AuthorizeAuthorizeContext, config oauth.OauthConfig, serviceConfig LoginServiceConfiguration) error {
+	redirectURL, err := keycloak.beforeRedirectToLogin(ctx, &ctx.RedirectURI, ctx.APIClient, ctx.RequestData, config, serviceConfig)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
 	}
-
-	referrerStr := referrerURL.String() + "&linked=false"
-	log.Debug(ctx, map[string]interface{}{
-		"code":        code,
-		"referrerURL": referrerURL.String(),
-		"user_name":   identity.Username,
-		"linked":      linked,
-	}, "linking identities...")
-
-	return keycloak.autoLinkProvidersDuringLogin(ctx, request, keycloakToken.AccessToken, referrerStr)
-
+	ctx.ResponseData.Header().Set("Location", *redirectURL)
```

Comment:
This is what I was referring to above. Le't set the header and return the redirect in the controller. Here, just return the location URL.